src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShutdownTask.java

```diff
@@ -97,47 +105,40 @@ public TaskResult call() {
         boolean applicationException = false;
 
         try {
-            ShutdownReason localReason = reason;
-            List<Shard> latestShards = null;
-            /*
-             * Revalidate if the current shard is closed before shutting down the shard consumer with reason SHARD_END
-             * If current shard is not closed, shut down the shard consumer with reason LEASE_LOST that allows active
-             * workers to contend for the lease of this shard.
-             */
-            if(localReason == ShutdownReason.TERMINATE) {
-                ShardClosureVerificationResponse shardClosureVerificationResponse = kinesisProxy.verifyShardClosure(shardInfo.getShardId());
-                if (shardClosureVerificationResponse instanceof ShardListWrappingShardClosureVerificationResponse) {
-                    latestShards = ((ShardListWrappingShardClosureVerificationResponse)shardClosureVerificationResponse).getLatestShards();
-                }
+            LOG.debug("Invoking shutdown() for shard " + shardInfo.getShardId() + ", concurrencyToken: "
+                              + shardInfo.getConcurrencyToken() + ", Shutdown reason: " + reason);
 
-                // If shard in context is not closed yet we should shut down the ShardConsumer with Zombie state
-                // which avoids checkpoint-ing with SHARD_END sequence number.
-                if(!shardClosureVerificationResponse.isShardClosed()) {
```

Comment:
If we're no longer verifying shard closure, what should we do in the case of ZOMBIE shards? How will the lease get dropped?

Reply:
ShutdownReason of ZOMBIE is caused by lost of lease. So if a shutdown task is triggered by this reason, the lease has already been lost. We don't need to lose it proactively. The reason that we used to have this change is that: we used to have false shardEnd event, where a null nextIterator is returned while the shard is not closed. So we do this validation, if the shard is not closed, we change the shutdown reason from TERMINATE to SHARD_END. Then we manually lose the lease to allow other workers to take it. But since now we are already doing the validation in the data retrieval layer, we can trust this shutdown reason in the shutdown task now.

Reply:
Got it, that makes sense. Thanks for clarifying!